src/features/search/useSearch.ts

```diff
@@ -0,0 +1,40 @@
+import { useAtomValue } from 'jotai/utils';
+import { apiClientsState } from '@store/api-clients';
+import { FoundResult, NotFoundResult } from '@common/types/search-results';
+import { useQuery } from 'react-query';
+import { useAppSelector } from '@common/state/hooks';
+import { selectSearchTerm } from '@features/search/searchSlice';
+import { useDebounce } from '@common/hooks/useDebounce';
+
+export const useSearch = () => {
+  const searchTerm = useAppSelector(selectSearchTerm);
+  const debouncedSearchTerm = useDebounce(searchTerm, 500);
+  const { searchApi } = useAtomValue(apiClientsState);
+  const id = searchTerm ? debouncedSearchTerm : '';
+  const query = useQuery(
+    ['search', id],
+    async () => {
+      try {
+        const result = await searchApi.searchById({ id });
+        return result as FoundResult;
+      } catch (e) {
+        try {
```

Comment:
I think this error handling logic could be extracted as we might use it more and more later.

Reply:
+1, I'd go further and suggest two hooks. This one, a higher-level abstraction used in the app. Plus one that handles just the `useSearchQuery` that returns all the react query options. [wallet example](https://github.com/hirosystems/stacks-wallet-web/tree/dev/src/app/query/mempool)

Reply:
Also file name introduces inconsistency

Reply:
> +1, I'd go further and suggest two hooks. This one, a higher-level abstraction used in the app. Plus one that handles just the `useSearchQuery` that returns all the react query options. [wallet example](https://github.com/hirosystems/stacks-wallet-web/tree/dev/src/app/query/mempool)

@kyranjamie what's the motive behind moving the query to a different file if it's only used here anyway (and the file size is small)?

Reply:
I'm not so concerned about it being a different file/same file. Motivation is more code reuse/SRP/loose coupling of the hook.

This hook:
- Interfaces with the store
- fetches data from the api

These could be thought of as distinct functionalities. Consider we want to reuse the search query hook elsewhere, and that it shouldn't touch an atom. We can't without refactoring.

Better that `useSearch` composes the functionality of smaller hooks, than putting everything in one big one.